source/common/router/rds_impl.cc

```diff
@@ -264,29 +241,36 @@ RouteConfigProviderPtr RouteConfigProviderManagerImpl::createStaticRouteConfigPr
   return provider;
 }
 
-ProtobufTypes::MessagePtr RouteConfigProviderManagerImpl::dumpRouteConfigs() {
+std::unique_ptr<envoy::admin::v2alpha::RoutesConfigDump>
+RouteConfigProviderManagerImpl::dumpRouteConfigs() const {
   auto config_dump = std::make_unique<envoy::admin::v2alpha::RoutesConfigDump>();
 
-  for (const auto& provider : getRdsRouteConfigProviders()) {
-    auto config_info = provider->configInfo();
-    if (config_info) {
-      auto* dynamic_config = config_dump->mutable_dynamic_route_configs()->Add();
-      dynamic_config->set_version_info(config_info.value().version_);
-      dynamic_config->mutable_route_config()->MergeFrom(config_info.value().config_);
-      TimestampUtil::systemClockToTimestamp(provider->lastUpdated(),
-                                            *(dynamic_config->mutable_last_updated()));
+  for (const auto& element : route_config_subscriptions_) {
+    // Because the RouteConfigProviderManager's weak_ptrs only get cleaned up
+    // in the RdsRouteConfigSubscription destructor, and the single threaded nature
+    // of this code, locking the weak_ptr will not fail.
+    auto subscription = element.second.lock();
+    ASSERT(subscription);
+    ASSERT(subscription->route_config_providers_.size() > 0);
+
+    auto* dynamic_config = config_dump->mutable_dynamic_route_configs()->Add();
+    if (subscription->config_info_) {
+      dynamic_config->set_version_info(subscription->config_info_.value().last_config_version_);
+      dynamic_config->mutable_route_config()->MergeFrom(subscription->route_config_proto_);
     }
+    TimestampUtil::systemClockToTimestamp(subscription->last_updated_,
```

Comment:
if subscription did not receive its first config (config_info_ is empty case),  I think  we don't need to create a config_dump entry.

Reply:
I did this change intentionally as you want to know the RDS config is delivered but not received route from RDS. And it is easier to track how many RDS subscription Envoy has from config dump, thoughs?

Reply:
OK,  it will be useful to know there is such subscriptions that did not even receive the first responses.

Reply:
Are you sure last_updated_ actually points to something meaningful here? Is it going to be 0 or the equivalent epoch? I guess I'm asking what will this actually output. :) 

My preference would actually be to do this change in a follow up as a discrete change, and we should be consistent across all the xDS? Thoughts?

Reply:
+1, prefer behavioral changes like this to be split out.

Reply:
It will be initialized when the subscription started:
https://github.com/envoyproxy/envoy/blob/028387a3b0746deaf011ea50104692dfbb8b8d2f/source/common/router/rds_impl.cc#L60

Sure, I will make a separate PR.

Reply:
I reverted this part for this PR.